core/src/test/java/apoc/cypher/CypherTest.java

```diff
@@ -158,6 +161,36 @@ public void testWithTimeout() {
                 result -> result.hasNext()));
     }
 
+    @Test
+    public void testRunTimeboxedWithTermination() {
+        final String query = "CALL apoc.cypher.runTimeboxed('unwind range (0, 10) as id CALL apoc.util.sleep(2000) return 0', null, 20000)";
+        checkTerminationGuard(db, query);
+    }
+
+    @Test
+    public void testRunTimeboxedWithTerminationInnerTransaction() {
+        final String innerLongQuery = "CALL apoc.util.sleep(10999) RETURN 0";
+        final String query = "CALL apoc.cypher.runTimeboxed($innerQuery, null, 99999)";
+
+        terminateTransactionAsync(db, 10L, innerLongQuery);
+        
+        // assert query terminated (RETURN 0)
+        TestUtil.testCall(db, query,
+                Map.of("innerQuery", innerLongQuery),
+                row -> assertEquals(Map.of("0", 0L), row.get("value")));
```

Comment:
Is there a way to check warnings returned maybe? (if there is one returned that is)

Reply:
In this case the `log.warn("query " + cypher + " has been terminated")` is not returned, because we terminate the inner transaction, so it doesn't execute that piece of code

Reply:
Ah I see, thanks :)

Reply:
and afaik there is no way for procedures to add a warning to the Cypher ExecutionContext